src/Core/Validation/ValidationResult.php

```diff
@@ -280,4 +263,25 @@ public function __unserialize(array $data): void
         $this->messages = $data['messages'];
         $this->isValid = $data['isValid'];
     }
+
+    /**
+     * Combine the data class and record ID from another ValidationResult object.
+     */
+    private function combineDataClassAndRecordID(ValidationResult $other): void
+    {
+        $thisDataClass = $this->getDataClass();
+        $otherDataClass = $other->getDataClass();
+        $thisRecordID = $this->getRecordID();
+        $otherRecordID = $other->getRecordID();
+        if ($thisDataClass === '' && $otherDataClass !== '') {
+            $this->setDataClass($otherDataClass);
+        } elseif ($thisDataClass !== '' && $otherDataClass === '') {
+            $other->setDataClass($thisDataClass);
+        }
+        if ($thisRecordID === 0 && $otherRecordID !== 0) {
+            $this->setRecordID($otherRecordID);
+        } elseif ($thisRecordID !== 0 && $otherRecordID === 0) {
+            $other->setRecordID($thisRecordID);
+        }
```

Comment:
We shouldn't be altering `$other`.
```suggestion
        if ($thisDataClass === '' && $otherDataClass !== '') {
            $this->setDataClass($otherDataClass);
        }
        if ($thisRecordID === 0 && $otherRecordID !== 0) {
            $this->setRecordID($otherRecordID);
        }
```

Reply:
Updated